### ISSUE TITLE
Adding debounce to `self.param.update`

### ISSUE THREAD

**Issue by ahuang11** [holoviz/param#984]
I would like a built-in mechanism in the `self.param.update` method (or similar) that supports debouncing. https://www.geeksforgeeks.org/debouncing-in-javascript/

This feature could be particularly useful for applications with frequent, rapid user interactions that trigger parameter updates (e.g., sliders or live data streams). It would improve the UX by removing flicker.

Use case would be linking widgets with `self.state.param.busy`, but at the current moment it could cause flickering:
https://github.com/holoviz/lumen/pull/783#discussion_r1868152821

I think it would be implemented here https://github.com/holoviz/param/blob/main/param/parameterized.py#L2297-L2357

**Comment by hoxbro** [comment]
I don't think this functionality should live in Param. I do think we should have some way to make the busy toggle not flicker on/off with a debounce, though I think this functionality should live in Panel.

**Comment by maximlt** [comment]
I'm not sure what the right solution is for the Panel issue reported here.

I think however that adding debouncing to Param could be a good fit as Param supports async callbacks.  I'm not entirely sure why the suggestion is for `self.param.update`, I'd see it more as a new keyword added to `.param.watch`. 

```python
import panel as pn

w_search = pn.widgets.TextInput(name='Search ...')
p_resuls = pn.MarkDown()

async def expensive_search(event):
    p_resuls.value = await db.query(...)

w_search.param.watch(expensive_search, 'value_input', debounce=3000)
pn.Column(w_search, p_resuls).servable()
```

@ahuang11 I wouldn't be surprised if that's a feature that was already implemented somehow in Panel for its internal needs. If that's really the case, that could motivate even more so implementing it in Param (many of the features of Param were first implemented in Panel).

> I think it would be implemented here https://github.com/holoviz/param/blob/main/param/parameterized.py#L2297-L2357

@ahuang11 this link no longer points to the right lines.
